daemon/create_windows.go

```diff
@@ -2,20 +2,41 @@ package daemon
 
 import (
 	"fmt"
+	"strings"
 
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/volume"
 )
 
+// dedupConfigEnvVars deduplicates requested environment variables in the config
+// as they are case insensitive on Windows
+func dedupConfigEnvVars(config *containertypes.Config) {
+	deduped := make(map[string]string)
+	for _, v := range config.Env {
+		i := strings.Index(v, "=")
+		if i >= 0 {
+			name := strings.ToUpper(v[:i])
+			value := v[i+1:]
+			deduped[name] = value
+		}
+	}
+	config.Env = nil
+	for k, v := range deduped {
+		config.Env = append(config.Env, k+"="+v)
+	}
+}
```

Comment:
Should they actually be deduped, or should this be left to Windows / the shell to do this?

On Linux we don't de-duplicate env-vars, e.g.;

```
docker run -dit -e foo=bar -e foo=baz busybox
```

```
            "Env": [
                "foo=bar",
                "foo=baz",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
```

It's up to the shell to set `foo=bar` first, then `foo=baz` (which will override the previous value)